Automattic/wp-calypso
# Domains: Refactor breadcrumb component to use common

The Breadcrumb component is in the components library so it can be used
from the domain pages

#### Changes proposed in this Pull Request

* Use the Breadcrumb component from the components library
* Extract the header of the domain pages to a common component that will render the action buttons displayed in the header
* Delete the Breadcrumb component from the domain components as the common one will be used

#### Testing instructions

* Navigate to the following domain pages and make sure the header area composed of breadcrumb and action buttons is displayed as expected:

|Page | url|
|-------|------|
| all-sites           |`/domain/manage`|
| connect-domain-step           |`/domains/mapping/:site/setup/:domain`|
| domain-management/settings           |`/domains/manage/all/:siteid/edit/:domain`|
| dns-records         |`/domains/manage/all/:siteid/dns/:domain`|
| add-dns-records         |`/domains/manage/:siteid/add-dns-record/:domain`|
| edit-contact-info         |`/domains/manage/:siteid/edit-contact-info/:domain`|
| site-domains         |`/domains/manage/:siteid`|
| transfer-page         |`/domains/manage/:siteid/transfer/:domain`|
| transfer-to-other-user         |`/domains/manage/:siteid/transfer/other-user/:domain`|
| transfer-to-other-site         |`/domains/manage/:siteid/transfer/other-site/:domain`|

#### Before and After for the responsive behaviour

|Page|Before | After|
|--|-------|------|
|all-domains|    ![all-domains-BEFORE](https://user-images.githubusercontent.com/3519124/165785568-789286d8-1a6f-4826-a14b-a8e4622bbcd0.gif)  |    ![all-domains-AFTER](https://user-images.githubusercontent.com/3519124/165785598-bcdf870b-5051-456a-b59f-a67e4af3ad93.gif)  |
|site-domains|    ![site-domains-BEFORE](https://user-images.githubusercontent.com/3519124/165785680-9e5af418-539e-43e7-b71b-de7ac57f4c94.gif)|      ![site-domains-AFTER](https://user-images.githubusercontent.com/3519124/165785708-c88f650f-d266-488e-a142-24d7984dda68.gif)|
|dns-record|    ![dns-record-BEFORE](https://user-images.githubusercontent.com/3519124/165786057-06212dab-00d8-4e97-a065-d0e967eca022.gif) |   ![dns-redord-AFTER](https://user-images.githubusercontent.com/3519124/165785771-4cde7c35-4ffc-4cc9-a800-0ef45af4d3ec.gif)   |

Related to #61952
